contracts/test/DualOptionVesting.t.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "forge-std/Test.sol";
+import "../src/DualOptionVesting.sol";
+import {MockedERC20} from "./mocks/MockedERC20.sol";
+
+contract DualOptionVestingTest is Test {
+    DualOptionVesting vesting;
+    MockedERC20 token;
+
+    address owner;
+    address treasury;
+    address user1;
+    address user2;
+    address user3;
+
+    // Declare reusable arrays for batch operations
+    address[] users = new address[](2);
+    uint256[] amounts = new uint256[](2);
+
+    function setUp() public {
+        owner = address(this);
+        treasury = address(0xBEEF);
+        user1 = address(0xAAA1);
+        user2 = address(0xAAA2);
+        user3 = address(0xAAA3);
+
+        token = new MockedERC20();
+        vesting = new DualOptionVesting(address(token), treasury);
+
+        // Mint tokens to the test contract (owner) first
+        token.mint(owner, 1_000_000 ether);
+        
+        // Transfer tokens to the vesting contract
+        token.transfer(address(vesting), 800_000 ether);
+    }
+
+    function testRegisterBatch() public {
+        users[0] = user1;
+        users[1] = user2;
+        amounts[0] = 1000 ether;
+        amounts[1] = 2000 ether;
+
+        vesting.registerBeneficiaries(users, amounts);
+
+        (uint256 alloc,,,) = vesting.beneficiaries(user1);
+        assertEq(alloc, 1000 ether);
+    }
+
+    function testChooseOptionAInstant() public {
+        registerSingle(user1, 1000 ether);
+
+        vm.prank(user1);
+        vesting.chooseOption(true);
+
+        uint256 expected = 50 ether;
+        assertEq(token.balanceOf(user1), expected);
+        assertEq(token.balanceOf(treasury), 950 ether);
+    }
+
+    function testChooseOptionBDelayedFullClaim() public {
+        registerSingle(user2, 1000 ether);
+
+        vm.prank(user2);
+        vesting.chooseOption(false);
+
+        // Wait 91 days
+        vm.warp(block.timestamp + 91 days);
+
+        vm.prank(user2);
+        vesting.claimDelayed();
+
+        assertEq(token.balanceOf(user2), 1000 ether);
+    }
+
+    function testChooseOptionBDelayedPartialClaim() public {
```

Comment:
We could be at least a bit more thorough with testing
Off the top of my head we're missing:
- Claim delayed instantly
- Claim delayed twice or more
- Forfeit works (claim instant then expect revert claim delayed) 
- The reverse of the last one above